pkg/providers/vsphere/vsphere.go

```diff
@@ -1081,7 +1081,7 @@ func buildTemplateMapCP(clusterSpec *cluster.Spec, datacenterSpec v1alpha1.VSphe
 
 	if clusterSpec.Spec.ProxyConfiguration != nil {
 		values["proxyConfig"] = true
-		var noProxyList []string
+		noProxyList := make([]string, 4)
```

Comment:
Why 4?
shouldn't it be the addition of all cidr blocks and noproxy lengths +1?
Also, when using append, I always find safer/cleaner using `make([]string, 0, 4)`. That would make sure the code doesn't break if for whatever reason we end up appending less that 4 items